tests/flakes.sh

```diff
@@ -845,3 +845,31 @@ nix store delete $(nix store add-path $badFlakeDir)
 
 [[ $(nix path-info      $(nix store add-path $flake1Dir)) =~ flake1 ]]
 [[ $(nix path-info path:$(nix store add-path $flake1Dir)) =~ simple ]]
+
+# Test nested flake overrides
+
+cat <<EOF > $flakeFollowsD/flake.nix
+{ outputs = _: {}; }
+EOF
+cat <<EOF > $flakeFollowsC/flake.nix
+{
+  inputs.D.url = "path:nosuchflake";
+  outputs = _: {};
+}
+EOF
+cat <<EOF > $flakeFollowsB/flake.nix
+{
+  inputs.C.url = "path:../flakeC";
+  outputs = _: {};
+}
+EOF
+cat <<EOF > $flakeFollowsA/flake.nix
+{
+  inputs.B.url = "path:./flakeB";
+  inputs.D.url = "path:./flakeD";
+  inputs.B.inputs.C.inputs.D.follows = "D";
+  outputs = _: {};
+}
+EOF
+
+nix flake lock $flakeFollowsA
```

Comment:
I get the idea of the test (the flake lock can only succeed if the override is properly taken into account), but it would be nice to have a more explicit check, if only to make the intent clearer. Maybe something like

```
nix flake metadata $flakeFollowsA --json | jq --exit-status '.locks.nodes.C.inputs.D == [ "D" ]'
```

Reply:
done in the same style as in the other tests in the file